src/contexts/UserContext.js

```diff
@@ -0,0 +1,56 @@
+import React, { createContext, useReducer, useContext } from "react";
+
+export const UserContext = createContext();
+
+// Initial state
+const initialState = {
+  isLoggedIn: false,
+  user: null,
+  error: null
+};
+
+// Actions
+export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
+export const LOGIN_FAIL = "LOGIN_FAIL";
+export const LOGOUT = "LOGOUT";
+
+// Action creators
+export function loginSuccess(user) {
+  return { type: LOGIN_SUCCESS, user };
+}
+
+export function loginFail(error) {
+  return { type: LOGIN_FAIL, error };
+}
+
+export function logout() {
+  return { type: LOGOUT };
+}
+
+// Reducer
+export function authReducer(state, action) {
+  switch (action.type) {
+    case LOGIN_SUCCESS:
+      return { isLoggedIn: true, user: action.user, error: null };
```

Comment:
To ensure that we don't lose any data in the state, perhaps we should use the spread operator?
return { ...state, isLoggedIn: true, user: action.user, error: null }